modules/rte_mis_school/src/Controller/FailedBulkDownload.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Drupal\rte_mis_school\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Datetime\DateFormatterInterface;
+use PhpOffice\PhpSpreadsheet\Spreadsheet;
+use PhpOffice\PhpSpreadsheet\Writer\Xlsx;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Returns responses for dblog routes.
+ */
+class FailedBulkDownload extends ControllerBase {
+
+  /**
+   * The database service.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $database;
+
+  /**
+   * The date formatter service.
+   *
+   * @var \Drupal\Core\Datetime\DateFormatterInterface
+   */
+  protected $dateFormatter;
+
+  /**
+   * The user storage.
+   *
+   * @var \Drupal\user\UserStorageInterface
+   */
+  protected $userStorage;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('database'),
+      $container->get('date.formatter'),
+    );
+  }
+
+  /**
+   * Constructs a DbLogController object.
+   *
+   * @param \Drupal\Core\Database\Connection $database
+   *   A database connection.
+   * @param \Drupal\Core\Datetime\DateFormatterInterface $date_formatter
+   *   The date formatter service.
+   */
+  public function __construct(Connection $database, DateFormatterInterface $date_formatter) {
+    $this->database = $database;
+    $this->dateFormatter = $date_formatter;
+    $this->userStorage = $this->entityTypeManager()->getStorage('user');
+  }
+
+  /**
+   * Displays details about a specific database log message.
+   *
+   * @param int $event_id
+   *   Unique ID of the database log message.
+   *
+   * @return \Symfony\Component\HttpFoundation\Response
+   *   Response object with Excel file content.
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
+   *   If no event found for the given ID.
+   */
+  public function widDetail($event_id) {
+    $dblog = $this->database->query('SELECT [w].*, [u].[uid] FROM {watchdog} [w] LEFT JOIN {users} [u] ON [u].[uid] = [w].[uid] WHERE [w].[wid] = :id', [':id' => $event_id])->fetchObject();
+    if (empty($dblog)) {
+      throw new NotFoundHttpException();
+    }
```

Comment:
Instead of `->query` can we use `->select` and the left join & right join functions?

Reply:
Sure.